server/src/main/java/com/google/coffeehouse/common/StorageHandler.java

```diff
@@ -62,29 +131,82 @@ public static String getPersonQuery(DatabaseClient dbClient, String userId) {
+          if (!resultSet.isNull("pronouns") || !resultSet.getString("pronouns").isEmpty()) {
+            pronouns = resultSet.getString("pronouns");
+          }
         }
-        personInfo = String.format("User ID: %s || Email: %s || Nickname: %s || %s\n",
-                                      userId, email, nickname, pronouns);
       }
+      Person.Builder personBuilder = Person.newBuilder(email, nickname);
+      if (!pronouns.isEmpty()) {
+        personBuilder.setPronouns(pronouns);
+      }
+      person = personBuilder.build();
     } else if (resultCount > 1) {
-      personInfo = ERROR_MORE_THAN_ONE_PERSON;
+      throw new IllegalArgumentException(ERROR_MORE_THAN_ONE_PERSON);
     } else {
-      personInfo = PERSON_DOES_NOT_EXIST;
+      throw new IllegalArgumentException(PERSON_DOES_NOT_EXIST);
     }
-    return personInfo;
+    return person;
   }
 
   /**
-  * Returns a string containing information about a club queried from the database.
-  * This method formats a string with a club's ID, book ID, description, name, and ownerID.
+  * Creates and returns a {@link Book} with the info about a book queried from the database.
+  * This method builds a {@link Book} with a book's ID, author, ISBN, and title.
+  * @param  dbClient  the database client
+  * @param  bookId    the book ID string used to query and get a book's information
+  * @return book      the Book object built containing the book information
+  */
+  public static Book getBookQuery(DatabaseClient dbClient, String bookId) {
+    Book book;
+    String author = "";
+    String isbn = "";
+    String title = "";
+    long resultCount = StorageHandlerHelper.getBookCountQuery(dbClient, bookId);
+    Statement statement = 
+        Statement.newBuilder(
+                "SELECT bookId, author, isbn, title "
+                  + "FROM Books "
+                  + "WHERE bookId = @bookId")
+            .bind("bookId")
+            .to(bookId)
+            .build();
+    if (resultCount == 1) {
+      try (ResultSet resultSet = dbClient.singleUse().executeQuery(statement)) {
+        while (resultSet.next()) {
+          title = resultSet.getString("title");
+          if (!resultSet.isNull("author") || !resultSet.getString("author").isEmpty()) {
+            author = resultSet.getString("author");
+          }
+          if (!resultSet.isNull("isbn") || !resultSet.getString("isbn").isEmpty()) {
+            isbn = resultSet.getString("isbn");
+          }
+        }
+      }
+      Book.Builder bookBuilder = Book.newBuilder(title);
+      if (!author.isEmpty()) {
+        bookBuilder.setAuthor(author);
+      }
+      if (!isbn.isEmpty()) {
+        bookBuilder.setAuthor(isbn);
+      }
+      book = bookBuilder.build();
+    } else if (resultCount > 1) {
+      throw new IllegalArgumentException(ERROR_MORE_THAN_ONE_BOOK);
+    } else {
+      throw new IllegalArgumentException(BOOK_DOES_NOT_EXIST);
+    }
+    return book;
```

Comment:
Completely optional, but it would save some lines in this function if you were to throw the `IllegalArgumentExceptions` directly after you determine `resultCount`. You would no longer need the `if (resultCount == 1)`, you could directly return `bookBuilder.build()`, and you could avoid declaring `Book book;`. Same thing goes for `getClubQuery`.

Reply:
Thanks for the suggestion, I like the code structured that way better!